jersey/src/main/java/com/netflix/conductor/server/resources/GenericExceptionMapper.java

```diff
@@ -62,24 +60,34 @@ public GenericExceptionMapper(Configuration config) {
 	}
 	
 	@Override
-	public Response toResponse(Throwable t) {
-		logger.error(t.getMessage(), t);
+	public Response toResponse(Throwable exception) {
+		logger.error(exception.getMessage(), exception);
 		Monitors.error("error", "error");
-		ApplicationException e = new ApplicationException(Code.INTERNAL_ERROR, t.getMessage(), t);
+
+		ApplicationException applicationException = null;
+
+        if (exception instanceof IllegalArgumentException) {
+            applicationException = new ApplicationException(Code.INVALID_INPUT, exception.getMessage(), exception);
```

Comment:
exception.getMessage() --> exception.toString(), e.g. for NPE, exception.getMessage() would give you 'null' while exception.toString() will give you `java.lang.NullPointerException`